### PR TITLE
fix permissions for the /stargate folder

### PR DESCRIPTION
**Pull request by ivansenic**
**What this PR does**:
1. Fixes the permissions for the `/stargate` folder and assigns it to the new `stargate` group and user
2. Adds executable flag to the `build.sh` files

### CODE REVIEW

File: cassandra-3.11/Dockerfile [hunk 1]
```diff
@@ -12,6 +12,9 @@ RUN apt update -qq \
     && apt clean all \
     && rm -rf /var/lib/apt/lists/*
 
+RUN mkdir /stargate \
```
Comment (jeffreyscarpenter) [inline cassandra-3.11/Dockerfile:15]: This step isn't needed. The `-d /stargate` in line 5 creates the directory with the correct permissions. Same with other files.

**Review by jeffreyscarpenter — CHANGES_REQUESTED** [review]

File: cassandra-3.11/Dockerfile (hunk 1 above)
Comment (ivansenic) [reply cassandra-3.11/Dockerfile:15]: Not true, it created the folder, but with root being the owner.. I checked by adding `RUN ls -all /` after the WORKDIR command..

Comment (ivansenic) [reply cassandra-3.11/Dockerfile:15]: Here's the proof: 

with this diff:
```
diff --git a/cassandra-3.11/Dockerfile b/cassandra-3.11/Dockerfile
index f38f4af..39b8816 100644
--- a/cassandra-3.11/Dockerfile
+++ b/cassandra-3.11/Dockerfile
@@ -15,6 +15,8 @@ RUN apt update -qq \
 USER stargate
 WORKDIR /stargate
 
+RUN ls -all /
+
 RUN curl -L https://github.com/stargate/stargate/releases/download/${STARGATE_VERSION}/stargate-jars.zip > stargate-jars.zip \
     && unzip stargate-jars -x "*/persistence-*" \
     && unzip stargate-jars "*/persistence-api*" \
```

I get:

```
$ docker build --tag stargateio/stargate-3_11:v1.0.52 .
Sending build context to Docker daemon  3.584kB
Step 1/10 : FROM openjdk:8-slim
 ---> 6ed55024947f
Step 2/10 : RUN set -x     && groupadd -r stargate --gid=999     && useradd -r -g stargate -d /stargate --uid=999 stargate
 ---> Using cache
 ---> 10800f8f98eb
Step 3/10 : ARG STARGATE_VERSION=v1.0.52
 ---> Using cache
 ---> bf64141da0e1
Step 4/10 : RUN apt update -qq     && apt install curl unzip iproute2 libaio1 -y     && apt autoremove --yes     && apt clean all     && rm -rf /var/lib/apt/lists/*
 ---> Using cache
 ---> 29103616b7ed
Step 5/10 : USER stargate
 ---> Using cache
 ---> 5ae41b96f13b
Step 6/10 : WORKDIR /stargate
 ---> Using cache
 ---> 68d536ce153a
Step 7/10 : RUN ls -all /
 ---> Running in d380550ac072
total 76
drwxr-xr-x   1 root root 4096 Mar 24 14:04 .
drwxr-xr-x   1 root root 4096 Mar 24 14:04 ..
-rwxr-xr-x   1 root root    0 Mar 24 14:04 .dockerenv
drwxr-xr-x   1 root root 4096 Mar 22 08:50 bin
drwxr-xr-x   2 root root 4096 Dec 11 17:25 boot
drwxr-xr-x   5 root root  340 Mar 24 14:04 dev
drwxr-xr-x   1 root root 4096 Mar 24 14:04 etc
drwxr-xr-x   2 root root 4096 Dec 11 17:25 home
drwxr-xr-x   1 root root 4096 Mar 16 00:00 lib
drwxr-xr-x   2 root root 4096 Mar 16 00:00 lib64
drwxr-xr-x   2 root root 4096 Mar 16 00:00 media
drwxr-xr-x   2 root root 4096 Mar 16 00:00 mnt
drwxr-xr-x   2 root root 4096 Mar 16 00:00 opt
dr-xr-xr-x 481 root root    0 Mar 24 14:04 proc
drwx------   1 root root 4096 Mar 17 18:16 root
drwxr-xr-x   3 root root 4096 Mar 16 00:00 run
drwxr-xr-x   1 root root 4096 Mar 22 08:50 sbin
drwxr-xr-x   2 root root 4096 Mar 16 00:00 srv
drwxr-xr-x   2 root root 4096 Mar 22 08:50 stargate
dr-xr-xr-x  13 root root    0 Mar 24 14:04 sys
drwxrwxrwt   1 root root 4096 Mar 22 08:50 tmp
drwxr-xr-x   1 root root 4096 Mar 16 00:00 usr
drwxr-xr-x   1 root root 4096 Mar 16 00:00 var
Removing intermediate container d380550ac072
 ---> bd67d1888a37
Step 8/10 : RUN curl -L https://github.com/stargate/stargate/releases/download/${STARGATE_VERSION}/stargate-jars.zip > stargate-jars.zip     && unzip stargate-jars -x "*/persistence-*"     && unzip stargate-jars "*/persistence-api*"     && unzip stargate-jars "*/persistence-cassandra-3.11*"     && rm stargate-jars.zip
 ---> Running in 90f12f99b99a
/bin/sh: 1: cannot create stargate-jars.zip: Permission denied
The command '/bin/sh -c curl -L https://github.com/stargate/stargate/releases/download/${STARGATE_VERSION}/stargate-jars.zip > stargate-jars.zip     && unzip stargate-jars -x "*/persistence-*"     && unzip stargate-jars "*/persistence-api*"     && unzip stargate-jars "*/persistence-cassandra-3.11*"     && rm stargate-jars.zip' returned a non-zero code: 2

```

Comment (jeffreyscarpenter) [reply cassandra-3.11/Dockerfile:15]: (Happened to use 4.0 image instead of 3.11, but results are the same)

Comment (jeffreyscarpenter) [reply cassandra-3.11/Dockerfile:15]: Not sure what could be the cause of our different results? I guess we could add in this step as long as we are sure that it is not going to fail the docker build in the case where the directory was already successfully created by the `useradd` command. Seems a shame to have to add an extra layer to the Docker container for this.

Comment (ivansenic) [reply cassandra-3.11/Dockerfile:15]: this has to be the difference between the `docker build` and `docker buildx` commands.. The second is used in the automatic release and works also by me.. But it feels bad that somebody comes here, tries to build the image and fails with the stupid permission denied error..

Comment (jeffreyscarpenter) [reply cassandra-3.11/Dockerfile:15]: Can we close this PR @ivansenic ?

Comment (ivansenic) [reply cassandra-3.11/Dockerfile:15]: My previous comment explains the situation.. If you think that's no issue, please close..

**Pull request closed without merging**